go/processor/floria/calls_test.go

```diff
@@ -0,0 +1,182 @@
+			},
+			success: true,
+		},
+		"failed": {
+			setup: func(interpreter *tosca.MockInterpreter) {
+				interpreter.EXPECT().Run(gomock.Any()).Return(tosca.Result{Success: false}, nil)
+			},
+			success: false,
+		},
+		"output": {
+			setup: func(interpreter *tosca.MockInterpreter) {
+				interpreter.EXPECT().Run(gomock.Any()).Return(tosca.Result{Success: true, Output: []byte("some output")}, nil)
+			},
+			success: true,
+			output:  []byte("some output"),
+		},
+	}
+
+	transaction := tosca.Transaction{
+		Sender:    tosca.Address{1},
+		Recipient: &tosca.Address{2},
+	}
+
+	ctrl := gomock.NewController(t)
+	context := tosca.NewMockTransactionContext(ctrl)
+	interpreter := tosca.NewMockInterpreter(ctrl)
+
+	senderBalance := tosca.NewValue(0)
+	recipientBalance := tosca.NewValue(0)
+
+	for name, test := range tests {
+		t.Run(name, func(t *testing.T) {
+			context.EXPECT().GetCodeHash(*transaction.Recipient).Return(tosca.Hash{})
+			context.EXPECT().GetCode(*transaction.Recipient).Return([]byte{})
+			context.EXPECT().CreateSnapshot()
+			context.EXPECT().GetBalance(transaction.Sender).Return(senderBalance)
+			context.EXPECT().SetBalance(transaction.Sender, senderBalance)
+			context.EXPECT().GetBalance(*transaction.Recipient).Return(recipientBalance)
+			context.EXPECT().SetBalance(*transaction.Recipient, recipientBalance)
+			context.EXPECT().RestoreSnapshot(gomock.Any()).AnyTimes()
+
+			test.setup(interpreter)
+
+			result, err := Call(interpreter, transaction, context)
+			if err != nil {
+				t.Errorf("Call returned an unexpected error: %v", err)
+			}
+			if result.Success != test.success {
+				t.Errorf("Unexpected success value from interpreter call")
+			}
+			if string(result.Output) != string(test.output) {
+				t.Errorf("Unexpected output value from interpreter call")
+			}
+		})
+	}
+}
+
+func TestProcessor_TransferValue(t *testing.T) {
+	transaction := tosca.Transaction{
+		Sender:    tosca.Address{1},
+		Recipient: &tosca.Address{2},
+		Value:     tosca.NewValue(10),
+	}
+
+	ctrl := gomock.NewController(t)
+	context := tosca.NewMockTransactionContext(ctrl)
+
+	context.EXPECT().GetBalance(transaction.Sender).Return(tosca.NewValue(100))
+	context.EXPECT().GetBalance(*transaction.Recipient).Return(tosca.NewValue(0))
+	context.EXPECT().SetBalance(transaction.Sender, tosca.NewValue(90))
+	context.EXPECT().GetBalance(transaction.Sender).Return(tosca.NewValue(90))
+	context.EXPECT().SetBalance(*transaction.Recipient, tosca.NewValue(10))
+	context.EXPECT().GetBalance(*transaction.Recipient).Return(tosca.NewValue(10))
+
+	err := transferValue(transaction, context)
+	if err != nil {
+		t.Errorf("transferValue returned an error: %v", err)
+	}
+	if context.GetBalance(transaction.Sender).Cmp(tosca.NewValue(90)) != 0 {
+		t.Errorf("Sender balance was not decremented")
+	}
+	if context.GetBalance(*transaction.Recipient).Cmp(tosca.NewValue(10)) != 0 {
+		t.Errorf("Recipient balance was not incremented")
+	}
```

Comment:
These checks only check whether you set the expectations above right for those calls. They do not test anything in the `transfereValue` function, and can be removed.

Reply:
I already thought so, but wanted to test more than just the returned error. I removed them and the ones in the other tests.